stdlib/public/runtime/SectionDataStatic.cpp

```diff
@@ -0,0 +1,61 @@
+//===-- SectionDataStatic.cpp -----------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Implementation of functions to read data sections from static executable.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstdlib>
+#include <cstring>
+#include "swift/Runtime/Debug.h"
+#include "SectionData.h"
+
+
+#if defined SUPPORTS_STATIC_BINARIES
+
+// If creating a static binary using --static-executable,
+// gold will set these to the data sections via --defsym
+const void *__swift2_protocol_conformances_start = nullptr;
+const void *__swift2_type_metadata_start = nullptr;
+
+
+void
+swift::_swift_initializeCallbacksForSectionData(InspectArgs *inspectArgs) {
+   const char *name = inspectArgs->sectionName;
+   const void **sectionDataAddr = nullptr;
+
+   if (!strcmp(name, SWIFT_PROTOCOL_CONFORMANCES_SECTION)) {
+     sectionDataAddr = &__swift2_protocol_conformances_start;
+   } else if (!strcmp(name, SWIFT_TYPE_METADATA_SECTION)) {
+     sectionDataAddr = &__swift2_type_metadata_start;
+   } else {
+     fatalError(/* flags = */ 0, "Invalid section name: %s", name);
+   }
+
+   if (sectionDataAddr == nullptr || *sectionDataAddr == nullptr) {
+     fatalError(/* flags = */ 0, "Static data section `%s' is null", name);
+   }
+   auto blockAddr = reinterpret_cast<const uint8_t*>(sectionDataAddr);
+   auto blockSize = *reinterpret_cast<const uint64_t*>(blockAddr);
+   blockAddr += sizeof(blockSize);
+   inspectArgs->fnAddImageBlock(blockAddr, blockSize);
+}
+
+extern "C" int
+dladdr(const void* addr, Dl_info* info)
+{
+  return 0;
+}
```

Comment:
@jckarter Ahh I see, you mean have a separate initialise function for each section rather than having them all share the same one, that makes it a bit easier to understand. Do you think in the future there will be many more data sections? I noticed some other ones defined in swift_sections.S but they only seemed to be used by the unit test code